server/game/cards/08.6-SAT/ToGoForwardYouMustGoBack.js

```diff
@@ -0,0 +1,58 @@
+const DrawCard = require('../../drawcard.js');
+
+class ToGoForwardYouMustGoBack extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Shuffle hand',
+            phase: 'dominance',
+            condition: () => this.game.getPlayers().some(this.hasHand),
+            handler: () => {
+                this.remainingPlayers = this.game.getPlayersInFirstPlayerOrder();
+                this.selections = [];
+                this.proceedToNextStep();
+            }
+        });
+    }
+
+    hasHand(player) {
+        return player.hand.length > 0;
+    }
+
+    proceedToNextStep() {
+        if(this.remainingPlayers.length > 0) {
+            let currentPlayer = this.remainingPlayers.shift();
+            if(!this.hasHand(currentPlayer)) {
+                this.proceedToNextStep();
+                return true;
+            }
+
+            this.game.promptWithMenu(currentPlayer, this, {
+                activePrompt: {
+                    menuTitle: 'Shuffle your hand to draw 5 cards?',
+                    buttons: [
+                        { text: 'Yes', method: 'shuffleHand' },
+                        { text: 'No', method: 'cancel' }
+                    ]
+                },
+                source: this
+            });
+        }
+    }
+
+    shuffleHand(player) {
+        player.hand.forEach(card => player.moveCard(card, 'draw deck'));
```

Comment:
I *think* this line and the next should be rewritten as the following, to make sure the deck gets shuffled after cards are moved back.

```javascript
for(let card of player.hand) {
    player.moveCard(card, 'draw deck', {}, () => {
        this.controller.shuffleDrawDeck();
    });
}
```

It does mean the deck will get shuffled after each card moved back, but that shouldn't matter.